packages/core/src/components/tabs/tabs.tsx

```diff
@@ -353,6 +353,12 @@ export class Tabs extends AbstractPureComponent<TabsProps, TabsState> {
             return (
                 <TabTitle
                     {...child.props}
+                    // add margin-right as inline style to reduce chance of double spacing between tabs
+                    // after switching from margin-right to gap in #6834
+                    // some consumers may have targeted the default margin-right to override it and this
+                    // should wipe out their override and at least reset them to the default gap style
+                    // instead of applying both the gap style and whatever margin-right they have defined
+                    style={{ ...child.props.style, marginRight: 0 }}
```

Comment:
tried making this work for the non tab case as well but looks like that will have issues since the `style` prop is not always applied at the top level, going to back this out

<img width="1314" alt="Screenshot 2024-07-03 at 3 41 30 PM" src="https://github.com/palantir/blueprint/assets/14102129/ad775c53-9384-4f51-9277-ec2e84014590">
<img width="1314" alt="Screenshot 2024-07-03 at 3 41 30 PM" src="https://github.com/palantir/blueprint/assets/14102129/ad775c53-9384-4f51-9277-ec2e84014590">